### ISSUE TITLE
Support capturing nested unknown fields

### ISSUE THREAD

**Issue by avivpxi** [PerimeterX/marshmallow#15]
Following the discussion on [this issue](https://github.com/PerimeterX/marshmallow/issues/12), we want to be able to capture unknown fields in nested objects.

**Comment by avivpxi** [comment]
After resolving [12](https://github.com/PerimeterX/marshmallow/issues/12), [this playground link](https://go.dev/play/p/CXicEUzg8-R) produces the following output:
```
err: <nil>
map: map[known:foo nested:{Known:goo} unknown:boo]
struct: &{Known:foo Nested:{Known:goo}}
```
meaning, nested object fields contain concrete structs instead of `map[string]interface{}`, and thus they cannot contain unknown fields (`map: map[known:foo nested:{Known:goo} unknown:boo]`, in out example).

[Here](https://github.com/PerimeterX/marshmallow/issues/12#issuecomment-1221303130), I presented a possible solution to optionally require all nested objects in the map to be maps as well. After thinking about it again I'm not sure this is very usable. Let's take @andeke07 use case presented [here](https://github.com/PerimeterX/marshmallow/issues/12#issue-1339770737), to iterate the `eventInfo` we will need:
```go
result, _ := marshmallow.Unmarshal(body, &incoming)
data := result["eventInfo"]
dataMap, ok := data.(map[string]interface{})
if ok {
  for key := range dataMap {
    // ...
  }
}
```

Since we have to lookup and cast, this is not aligning with what Marshmallow tries to provide.
I can think of an alternative solution, allowing nested structs to customize how they handle map data by implementing a `HandleJSONData` method:

```go
type IncomingMessage struct {
	AlertSummary        string         `json:"alert_summary" validate:"required"`
	DetailedDescription string         `json:"detailed_description" validate:"required"`
	DeviceName          string         `json:"deviceName" validate:"required"`
	EventMetaData       *EventMetaData `json:"eventInfo" validate:"required"`
	Options             struct {
		AffectedCi   string `json:"affected_ci"`
		AffectedArea string `json:"affected_area"`
		HelpURL      string `json:"help_url"`
	} `json:"options"`
}

type EventMetaData struct {
	EventType string `json:"eventType" validate:"required"`
}

func (e *EventMetaData) HandleJSONData(data map[string]interface{}) {
	for key := range data {
		// ...
	}
}
```

**Comment by avivpxi** [comment]
@andeke07 would that be convenient/usable?

**Comment by andeke07** [comment]
@avivpxi thanks for this and also sorry for not replying on my other issue - I was away and not getting any email updates so I missed it. That looks like an interesting approach. So unknown fields in the root of the JSON would show up in the produced map, and unknown nested fields would be acted on via this method?

**Comment by avivpxi** [comment]
That was my thought, yes.
Actually, if the root object implements `HandleJSONData` it will also get called but you would be able to get them directly from the Unmarshall call like you do today.

Any other ideas?
Obviously, feel free to open a PR if you like to contribute. 🙏

**Comment by avivpxi** [comment]
@andeke07, this is resolved in v1.1.2
Showcased in [this playground link](https://go.dev/play/p/azUUjuN9oYz).
(ATM it might cache v1.1.1 and not show the fix but try running the code against the latest version)
